### ISSUE TITLE
Request: docs on using generated code, in particular, auth

### ISSUE THREAD

**Issue by shankiesan** [mikestead/openapi-client#16]
Most of the code that the package generates is pretty dead ahead, but for the life of me I can't figure out how to use my bearer token with Redux. [This line](https://github.com/mikestead/openapi-client/blob/cf3784b7c4e6134e0e42a5078b68cf691462d711/src/gen/js/service.js.template#L30) hints that there should be an `options.getAuthorization`, but I can't work out how to do that with my Redux action.

If you can give me some quick tips, I can then make a PR to update the docs. If I get the time, I'll also try to add a few more examples about how to use the generated code.

**Comment by mikestead** [comment]
Hey @shankiesan I appreciate the feedback and the kind offer, sorry this wasn't documented.

Here's a pure JS example (not sure if you're using TypeScript or not?). I've edit it down from a real world codebase so excuse any typos

```
import serviceGateway from './service/gateway';

serviceGateway.init({
  // set your service url explicitly or defaults to one you gen'd code from
  url: 'https://service.com/api', 
  // Add a `getAuthorization` handler for when a request requires auth credentials
  getAuthorization
});

// param 'security' represents the security definition in your openapi spec a request is requiring
// For bearer type it has two properties:
// 1. id - the name of the security definition
// 2. scopes - the token scope(s) required
// Should return a promise
function getAuthorization(security) {
  switch (security.id) {
    case 'account': return getAccountToken(security);
    default: throw new Error(`Unknown security type '${security.id}'`)
  }
};

function getAccountToken(security) {
  // some util to search tokens in redux state or local storage
  const token = findAccountToken(security);
  if (token) return Promise.resolve({ token: token.value });
  else throw new Error(`Token ${type} ${security.scopes} not available`);
}
```

**Comment by shankiesan** [comment]
Hey @mikestead I'm just vanilla JS, so this is perfect (and makes sense, at a first glance)! I'm under the cosh to get something working this morning, so this is perfect timing. 

I'll get my stuff working this morning and get back to you with any issues. Then, when I'm back from holiday in a fortnight, I'll submit a PR to update the docs.

I was explaining this project to one of my business partners who isn't that _au fait_ with the JS world, and he said "So it's like Sass for Swagger? Cool". Not exactly right, but exciting all the same.

**Comment by mikestead** [comment]
@shankiesan yeah let me know if you run into any problems or questions. And thanks again. I may get to those docs before you get back, but can't guarantee it as work etc etc.

**Comment by shankiesan** [comment]
Yeah that's cool. I'm more than happy to do it as I think it's a really worthwhile project, and it might be worth getting my perspective as a user that's not familiar with the inner workings...

**Comment by shankiesan** [comment]
I'm happy to report that that's all good. I'll have a go at the docs when I get back, if you don't do it first :)

**Comment by shankiesan** [comment]
Resolved in #18
